Upgrade to angular 19 release note (#3682)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1022-0-0-angular-19-upgrade-and-standalone-components-release-note.md

---

> date: 2025-06-16

Reviewer: @rahulc8y If we use this date, probably nobody will ever read this change log. Is this intended?

Reviewer 2: Since Angular 19 apps were released at the end of Q2, with a separate announcement on the Tech Community at that time, and coverage in the “What’s New” training through videos, using a current date for this changelog would be misleading given the public availability of these sources. That’s why, out of the two options—previous date or current date—I would prefer the older date.

Additionally, DTM published a changelog in August highlighting updates to WebSK using Angular 19, so using any date later than that would also be incorrect.

title: Angular 19 upgrade and standalone components
product_area: Application enablement & solutions
change_type:
  - value: change-inv-3bw8e
    label: Improvement
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-62856
version: 1022.0.0
---

> Starting with version 1022.0.0, the Web SDK has been upgraded to Angular 19, bringing performance improvements and new features. This update brings improvements and new features, but may also introduce breaking changes that could impact your existing implementations.

Reviewer: ```suggestion
Starting with version 1022.0.0, the Web SDK has been upgraded to Angular 19. This update brings performance improvements and new features, but may also introduce breaking changes that could impact your existing implementations.
```

> Comprehensive upgrade documentation is available to help developers navigate the transition and update their codebases accordingly- see the [Angular 19 Upgrade Guide](/web/upgrade/#upgrading-to-angular-19).

Reviewer: ```suggestion
Comprehensive upgrade documentation is available to help developers navigate the transition and update their codebases accordingly, see the [Angular 19 Upgrade Guide](/web/upgrade/#upgrading-to-angular-19).
```

To ensure a day one compatibility of your plugins with applications based on the v1022 Web SDK, you should add the standalone flag to all components/pipes/directives of your plugins.

> You do not need to migrate your components/pipes/directives to `standalone: true`, you can also set it to `false`. You just need to ensure that you've set the standalone flag.

Reviewer: ```suggestion
You do not have to migrate your components/pipes/directives to `standalone: true`, you can also set them to `false`. You just need to ensure that you've set the standalone flag.
```

> Angular v19 will otherwise default to `standalone: true` instead of `standalone: false` if the `standalone` flag has not been defined, which can break your components/pipes/directives.

Reviewer: ```suggestion
If the standalone flag has not been defined, Angular v19 will default to `standalone: true` instead of `standalone: false`, which can break your components/pipes/directives.
```
